### PR TITLE
docs on 6.0.1

### PR DESCRIPTION
**Pull request by cel055**
[DOG-713](https://cognitedata.atlassian.net/browse/DOG-713)

[DOG-713]: https://cognitedata.atlassian.net/browse/DOG-713?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: cognite/extractorutils/configtools/elements.py [hunk 1]
```diff
@@ -103,6 +112,10 @@ def either_id(self) -> EitherId:
 
 
 class TimeIntervalConfig(yaml.YAMLObject):
+    """
+    Configuration parameter for setting the time interval
```
Comment (mathialo) [inline cognite/extractorutils/configtools/elements.py:116]:
``` suggestion
    Configuration parameter for setting a time interval
```

File: cognite/extractorutils/configtools/elements.py [hunk 2]
```diff
@@ -167,6 +180,10 @@ def __repr__(self) -> str:
 
 
 class FileSizeConfig(yaml.YAMLObject):
+    """
+    Configuration parameter for setting the file size
```
Comment (mathialo) [inline cognite/extractorutils/configtools/elements.py:184]:
``` suggestion
    Configuration parameter for setting a file size
```

File: cognite/extractorutils/configtools/elements.py [hunk 3]
```diff
@@ -545,6 +562,12 @@ def stop_pushers(self) -> None:
 
 
 class ConfigType(Enum):
+    """
+    Enum to define if the configuration file should be fetched remotely.
+
+    Valid values: local|remote
```
Comment (mathialo) [inline cognite/extractorutils/configtools/elements.py:568]: I prefer for these to not be in docs, they are always visible from the code, and we run the risk of them falling out of sync.
``` suggestion
```

File: cognite/extractorutils/configtools/elements.py [hunk 4]
```diff
@@ -569,23 +592,39 @@ class BaseConfig(_BaseConfig):
...
+    """
+
     database: str
     table: str
 
 
 @dataclass
 class RawStateStoreConfig(RawDestinationConfig):
+    """
+    Configuration of the State Store based on CDF RAW
```
Comment (mathialo) [inline cognite/extractorutils/configtools/elements.py:606]:
``` suggestion
    Configuration of a state store based on CDF RAW
```

File: cognite/extractorutils/configtools/elements.py [hunk 5]
```diff
@@ -569,23 +592,39 @@ class BaseConfig(_BaseConfig):
...
+    """
+
     upload_interval: TimeIntervalConfig = TimeIntervalConfig("30s")
 
 
 @dataclass
 class LocalStateStoreConfig:
+    """
+    Configuration of the State Store when using locally
```
Comment (mathialo) [inline cognite/extractorutils/configtools/elements.py:615]:
``` suggestion
    Configuration of a state store using a local JSON file
```

File: cognite/extractorutils/configtools/loaders.py [hunk 6]
```diff
@@ -41,11 +41,21 @@
 
 
 class KeyVaultAuthenticationMethod(Enum):
+    """
+    Enum to define the access to Azure
+
+    Valid values: default|client-secret
```
Comment (mathialo) [inline cognite/extractorutils/configtools/loaders.py:47]:
``` suggestion
```

File: cognite/extractorutils/util.py [hunk 7]
```diff
@@ -197,24 +197,27 @@ def add_extraction_pipeline(
...
     """
-    This is to be used as a decorator for extractor functions to add extraction pipeline information
+    This is to be used as a decorator for extractor functions to add extraction pipeline information.
 
     Args:
-        extraction_pipeline_ext_id:
-        cognite_client:
-        heartbeat_waiting_time:
-        added_message:
+        extraction_pipeline_ext_id: Id of the extraction pipeline
```
Comment (mathialo) [inline cognite/extractorutils/util.py:203]:
``` suggestion
        extraction_pipeline_ext_id: External ID of the extraction pipeline
```

File: cognite/extractorutils/util.py [hunk 8]
```diff
@@ -197,24 +197,27 @@ def add_extraction_pipeline(
...
+        cognite_client: Assets to create
```
Comment (mathialo) [inline cognite/extractorutils/util.py:204]:
``` suggestion
        cognite_client: Client to use when communicating with CDF
```

File: cognite/extractorutils/util.py [hunk 9]
```diff
@@ -197,24 +197,27 @@ def add_extraction_pipeline(
...
+        heartbeat_waiting_time: Interval of the execution of the Thread
```
Comment (mathialo) [inline cognite/extractorutils/util.py:205]:
``` suggestion
        heartbeat_waiting_time: Target interval between heartbeats, in seconds
```

File: cognite/extractorutils/util.py [hunk 10]
```diff
@@ -197,24 +197,27 @@ def add_extraction_pipeline(
...
-        )
-        def extract_data(*args, **kwargs):
-            <INSERT FUNCTION BODY>
+
+        .. code-block:: python
+
+            @add_extraction_pipeline(
+                extraction_pipeline_ext_id=<INSERT EXTERNAL ID>,
+                cognite_client=<INSERT COGNITE CLIENT OBJECT>,
+                logger=<INSERT LOGGER>,
```
Comment (mathialo) [inline cognite/extractorutils/util.py:217]: I see this was from the earlier text as well, but it does not seem to me that we expect a logger argument here?

``` suggestion
```

File: docs/source/docs.rst [hunk 11]
```diff
@@ -75,14 +92,14 @@ Exceptions
...
-.. automodule:: cognite.extractorutils.throttle
-    :members:
-    :undoc-members:
-    :inherited-members:
-    :show-inheritance:
+.. .. automodule:: cognite.extractorutils.throttle
+..     :members:
+..     :undoc-members:
+..     :inherited-members:
+..     :show-inheritance:
```
Comment (mathialo) [inline docs/source/docs.rst:102]: I think we refactored this module out in a previous release, could you double check that and remove it if we did?

**Review by mathialo — COMMENTED** [review]
Nice, some nitpick

File: cognite/extractorutils/util.py (hunk 10 above)
Comment (cel055) [reply cognite/extractorutils/util.py:217]: Good catch, no there isn't, right now there is only the `added message` parameter, wich is a str

File: docs/source/docs.rst (hunk 11 above)
Comment (cel055) [reply docs/source/docs.rst:102]: It was removed, I left there just so you can see it and maybe tell me something that I don't know. I also left the `cognite.extractorutils.uploader_extractor`, commented, just so we know that there is something there but it's not documented, do you want me to remove?

Comment (mathialo) [reply docs/source/docs.rst]: I think we should just remove it

File: cognite/extractorutils/configtools/elements.py (hunk 1 above)
Comment (cel055) [reply cognite/extractorutils/configtools/elements.py]: done

File: cognite/extractorutils/configtools/elements.py (hunk 2 above)
Comment (cel055) [reply cognite/extractorutils/configtools/elements.py]: done

File: cognite/extractorutils/configtools/elements.py (hunk 4 above)
Comment (cel055) [reply cognite/extractorutils/configtools/elements.py]: done

File: cognite/extractorutils/configtools/elements.py (hunk 5 above)
Comment (cel055) [reply cognite/extractorutils/configtools/elements.py]: done

File: cognite/extractorutils/configtools/loaders.py (hunk 6 above)
Comment (cel055) [reply cognite/extractorutils/configtools/loaders.py]: done

File: cognite/extractorutils/util.py (hunk 7 above)
Comment (cel055) [reply cognite/extractorutils/util.py]: done

File: cognite/extractorutils/util.py (hunk 8 above)
Comment (cel055) [reply cognite/extractorutils/util.py]: done

File: cognite/extractorutils/util.py (hunk 10 above)
Comment (cel055) [reply cognite/extractorutils/util.py]: done

**Review by mathialo — APPROVED** [review]

**Pull request merged**